lib/packwerk/cache.rb

```diff
@@ -0,0 +1,131 @@
+# frozen_string_literal: true
+# typed: strict
+
+#
+# There are some known bugs in this cache:
+# 1) The cache should be busted if the contents of `packwerk.yml` change, since custom associations
+# can affect what is considered a violation
+# 2) The cache should be busted if inflections change.
```

Comment:
This sadly is a little difficult - we could assume all inflections to be defined in a certain file, but we would never cover all cases. For example, we found that some gems add inflections to the main app.

Reply:
On the other hand, now that we boot up the application for `packwerk check` anyway, instead of comparing file contents, could we save `application_inflections.hash` and save that, use it as a trigger to clear the cache?

Reply:
I wrote my above comment before reading this one. I think conceptually that could work.

That particular implementation I believe would not work:
https://stackoverflow.com/questions/57611448/why-is-ruby-hash-producing-different-output-every-time

Unless `Rails.application_inflections.hash` is implemented differently, `hash` is unstable across processes.